src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -985,6 +1022,8 @@ SDCPUThread::onInstDataFetched(weak_ptr<InflightInst> inst,
 
             decode_result = curMacroOp->fetchMicroop(pc.microPC());
 
+            if (!decode_result->isFirstMicroop()) --numInstsInWindow;
```

Comment:
So we subtract from `numInstsInWindow` for *all* microops except the first? This confuses me.

Maybe I don't understand what `numInstsInWindow` is.

Reply:
Sometimes a microcoded instruction will branch to a new pc within the same macroop which causes a fetch to occur, even though it never completed the prior one. This check is to undo the increment which happened in advanceInst in that case.

Reply:
Can you add this as a comment? The code is not self explanatory.

Reply:
done